builder/pipeline.cc

```diff
@@ -621,7 +621,11 @@ class pipeline_builder {
       for (int d = 0; d < static_cast<int>(o.dims.size()); ++d) {
         if (o.dims[d] == loop.sym()) {
           // This output uses this loop. Add it to the bounds.
-          bounds |= buffer_bounds(o.sym(), d);
+          if (allocation_bounds_[o.sym()]) {
+            bounds |= (*allocation_bounds_[o.sym()])[d];
```

Comment:
This seems like it's going in the opposite direction of #278 and #280 to use symbolic bounds more, which I guess I thought were improvements in those cases. Any intuition for why it's different here?

Perhaps this is a workaround for `slide_and_fold_storage` not tracking bounds well? Or the simplifier?

Reply:
I don't think it really does. The inferred allocation bounds still only use symbolic bounds without any substitions from what user have provided.